src/join.sol

```diff
@@ -318,6 +318,67 @@ contract GemJoin5 is LibNote {
     }
 }
 
+// GemJoin6
+// For a token with a proxy and implementation contract (like tUSD)
+//  If the implementation behind the proxy is changed, this prevents joins
+//   and exits until the implementation is reviewed and approved by governance.
+
+contract GemLike6 {
+    function decimals() public view returns (uint);
+    function balanceOf(address) public returns (uint256);
+    function transfer(address, uint256) public returns (bool);
+    function transferFrom(address,address,uint) public returns (bool);
+    function implementation() public view returns (address);
+}
+
+contract GemJoin6 is LibNote {
+    // --- Auth ---
+    mapping (address => uint) public wards;
+    function rely(address usr) external note auth { wards[usr] = 1; }
+    function deny(address usr) external note auth { wards[usr] = 0; }
+    modifier auth {
+        require(wards[msg.sender] == 1, "GemJoin/not-authorized");
+        _;
+    }
+
+    VatLike  public vat;
+    bytes32  public ilk;
+    GemLike6 public gem;
+    uint     public dec;
+    uint     public live;  // Access Flag
+
+    mapping (address => bool) public implementations;
```

Comment:
I'd use `(address => uint256)` just to keep the same behaviour than with `wards`.
Also it is a bit easier to do FV.

Reply:
Actually why not just allow one implementation? and rename it to `code`. Which could be an address or a bytes32 in case we prefer to use the `extcodehash` value

Reply:
Updated, we should push to fix FV. I much prefer bools where applicable.

Reply:
Well FV is still possible, just some masking needed, but it's like the whole system now relies on `uint256` for bools. But yes I agree with you that booleans look much better.